File: devops-agent-iac/anexos/cdk-typescript/lib/constructs/platform-contract.ts
import * as ssm from 'aws-cdk-lib/aws-ssm';
import { Construct } from 'constructs';

export interface PlatformContractProps {
  readonly parameterName: string;
  readonly payload: Record<string, unknown>;
}

/**
 * Publica o "contrato de plataforma" em SSM Parameter Store.
 * Os próximos agentes leem este parâmetro em vez de importar outputs de stack,
 * o que evita acoplamento rígido (CloudFormation exports bloqueiam alterações).
 */
export class PlatformContract extends Construct {
  public readonly parameter: ssm.StringParameter;

  constructor(scope: Construct, id: string, props: PlatformContractProps) {
    super(scope, id);
    this.parameter = new ssm.StringParameter(this, 'Parameter', {
      parameterName: props.parameterName,
      stringValue: JSON.stringify(props.payload),
      tier: ssm.ParameterTier.ADVANCED,
      description: 'Contrato da camada de infraestrutura base consumido pelos agentes seguintes',
    });
  }
}

File: devops-agent-iac/abordagem-2-cdk-python/app_lib/constructs/platform_contract.py
"""Publica o contrato da camada base em SSM Parameter Store."""
import json

from aws_cdk import aws_ssm as ssm
from constructs import Construct


class PlatformContract(Construct):
    def __init__(
        self, scope: Construct, construct_id: str, *, parameter_name: str, payload: dict
    ) -> None:
        super().__init__(scope, construct_id)
        self.parameter = ssm.StringParameter(
            self,
            "Parameter",
            parameter_name=parameter_name,
            string_value=json.dumps(payload),
            tier=ssm.ParameterTier.ADVANCED,
            description="Contrato da camada base consumido pelos agentes seguintes",
        )
